src/hooks.ts

```diff
@@ -16,47 +16,67 @@ const AUTH_RESULT_TYPE = "authResult";
 
 type MessageHandlerFunc = (e: MessageEvent) => void;
 
+export const useAuthState = (): AuthSliceState => useSelector((state: RootState) => state.auth);
+
 export const useIsAuthenticated = () => {
-    const { idTokenContents } = useSelector((state: RootState) => state.auth);
+    const { idTokenContents } = useAuthState();
     return getIsAuthenticated(idTokenContents);
 };
 
-export const useAccessToken = () => useSelector((state: RootState) => state.auth.accessToken);
+export const useAccessToken = () => useAuthState().accessToken;
 
 export const useAuthorizationHeader = () => {
     const accessToken = useAccessToken();
     return useMemo(() => makeAuthorizationHeader(accessToken), [accessToken]);
 };
 
-export const useResourcePermissions = (resource: Resource, authzUrl: string) => {
+export const useResourcesPermissions = (resources: Resource[], authzUrl: string) => {
     const dispatch: AppDispatch = useDispatch();
 
     const haveAuthorizationService = !!authzUrl;
 
-    const key = useMemo(() => makeResourceKey(resource), [resource]);
+    const keys = useMemo(() => resources.map((resource) => makeResourceKey(resource)), [resources]);
 
-    const { permissions, isFetching, hasAttempted, error } =
-        useSelector((state: RootState) => state.auth.resourcePermissions?.[key]) ?? {};
+    const { resourcePermissions } = useAuthState();
 
     useEffect(() => {
-        if (!haveAuthorizationService || isFetching || permissions || hasAttempted) return;
-        dispatch(fetchResourcePermissions({ resource, authzUrl }));
+        const allFetching = keys.reduce((acc, key) => acc || !!resourcePermissions[key]?.isFetching, false);
+        const allHavePermissions =
+            keys.reduce((acc, key) => acc && !!resourcePermissions[key]?.permissions?.length, true);
+        const allAttempted = keys.reduce((acc, key) => acc && !!resourcePermissions[key]?.hasAttempted, true);
```

Comment:
I think using `Array.every()` instead of `Array.reduce()` would be better here, since `every` will exit as soon as the condition is not satisfied.

Reply:
forgot about that one - gotta change some WIP bento web stuff too...

Reply:
Array.some is also good elsewhere in this diff